### PR TITLE
Added annotations to axiom-js

### PR DESCRIPTION
**Pull request by schehata**

### CODE REVIEW

File: integration/src/annotations.spec.ts [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    });
+  });
+
+  describe('update', () => {
+    it('should update the annotation', async () => {
+      const dataset = await client.update(id, {
+        description: 'This is a soon to be filled test dataset',
+      });
+
+      expect(dataset.description).toEqual('This is a soon to be filled test dataset');
```
Comment (bahlo) [inline integration/src/annotations.spec.ts:48]: the description is wrong

File: packages/js/package.json [hunk 2]
```diff
@@ -1,7 +1,7 @@
 {
     "name": "@axiomhq/js",
     "description": "The official javascript bindings for the Axiom API",
-    "version": "1.0.0-rc.4",
```
Comment (bahlo) [inline packages/js/package.json:4]: can probably do 1.1.0?

File: packages/js/src/annotations.ts [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        datasets: string[];
+        title?: string;
+        description?: string;
+        url?: string;
+        time?: string;
+        endTime?: string;
+    }
+
+    export interface UpdateRequest {
+        description: string;
```
Comment (bahlo) [inline packages/js/src/annotations.ts:33]: You cal update every field except id

File: packages/js/package.json (hunk 2 above)
Comment (schehata) [reply packages/js/package.json:4]: but we haven't released stable `1.0.0` yet

File: packages/js/src/annotations.ts (hunk 3 above)
Comment (schehata) [reply packages/js/src/annotations.ts]: updated

File: packages/js/package.json (hunk 2 above)
Comment (schehata) [reply packages/js/package.json:4]: will release `v1.0.0` as agreed.

File: integration/src/annotations.spec.ts (hunk 1 above)
Comment (schehata) [reply integration/src/annotations.spec.ts:52]: fixed

File: packages/js/src/annotations.ts [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+export namespace annotations {
+
+    export interface Annotation {
+        id: string;
+        type: string;
+        datasets: string[];
+        title?: string;
+        description?: string;
+        url?: string;
+        time?: string;
```
Comment (bahlo) [inline packages/js/src/annotations.ts:12]: Time is not optional here

File: packages/js/src/annotations.ts [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+    }
+
+    export interface UpdateRequest {
+        type: string;
+        datasets: string[];
+        title?: string;
+        description?: string;
+        url?: string;
+        time?: string;
+        endTime?: string;
```
Comment (bahlo) [inline packages/js/src/annotations.ts:39]: every field is optional here

Comment (schehata) [reply packages/js/src/annotations.ts:39]: hmmmm, I thought we don't do patch requests, but only PUT requests with the full object. That's why I thought the required fields for creation are required here as well.

File: packages/js/src/annotations.ts (hunk 4 above)
Comment (schehata) [reply packages/js/src/annotations.ts]: fixed

File: packages/js/src/annotations.ts (hunk 5 above)
Comment (schehata) [reply packages/js/src/annotations.ts:39]: updated

**Review by bahlo — APPROVED** [review]

**Pull request merged**